iroha-cli/interactive/impl/interactive_common_cli.cpp

```diff
@@ -79,13 +82,18 @@ namespace iroha_cli {
       return line;
     }
 
-    void printEnd() { std::cout << "--------------------" << std::endl; }
+    void printEnd() {
+      std::cout << "--------------------" << std::endl;
+    }
 
     nonstd::optional<std::pair<std::string, uint16_t>> parseIrohaPeerParams(
-        ParamsDescription params) {
-      auto address = params[0];
-      auto port = parser::parseValue<uint16_t>(params[1]);
-      if (not port.has_value()) {
+        ParamsDescription params,
+        const std::string &default_ip,
+        const int &default_port) {
+      auto address = params[0].empty() ? default_ip : params[0];
+      auto port = params[1].empty() ? default_port
+                                    : parser::parseValue<uint16_t>(params[1]);
+      if (not params.empty() and not port.has_value()) {
```

Comment:
Why do you access `params[0]` and `params[1]` and after checks `params` on emptiness? Looks like an UB

Reply:
This is the case when no input is provided to peer_ip and peer_port, aka use default values

Reply:
for address it is definitely better to write ```auto address = params[0].value_or(default_ip);```, where params[0] is optional

Reply:
This will not work, since `params[0]` is not optional